### PR TITLE
Config client

### PR DESCRIPTION
**Pull request by jcrichlake**
## Description

Adding a config parsing utility that will allow us to receive the texts of any config file and load it into a struct. 

## Issue

[Issue](https://github.com/CDCgov/trusted-intermediary/issues/1082)

## Checklist

- [x] I have added tests to cover my changes
- [x] I have added logging where useful (with appropriate log level)
- [x] I have updated the documentation accordingly

**Note**: You may remove items that are not applicable

### CODE REVIEW

File: src/utils/config.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	partnerSettings PartnerSettings
+	PartnerId       string //unique name, put in queue message from polling function - keep this short so we can use it in TF resources? Currently TF and RS use `ca-phl`
+}
+
+func (config *Config) populatePartnerSettings(input []byte) error {
+
+	var partnerSettings PartnerSettings
+	jsonData := input
+
+	err := json.Unmarshal(jsonData, &partnerSettings)
```
Comment (halprin) [inline src/utils/config.go:37]: Any reason for this rename?  You can just do this...

```suggestion
	err := json.Unmarshal(input, &partnerSettings)
```

File: src/utils/config.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	IsActive                 bool   `json:"isActive"`
+	IsExternalSftpConnection bool   `json:"isExternalSftpConnection"`
+	HasZipPassword           bool   `json:"hasZipPassword"`
+	DefaultEncoding          string `json:"defaultEncoding"`
+}
+
+// When did we last get the file and what did we parse out of it?
+type Config struct {
+	lastRetrieved   time.Time
+	partnerSettings PartnerSettings
```
Comment (halprin) [inline src/utils/config.go:28]: Just checking that you are purposely making these private fields (due to the lowercase start of the name).

**Review by halprin — COMMENTED** [review]
Mostly some clarifying questions.

File: src/utils/config_test.go
Comment (pluckyswan) [inline src/utils/config_test.go:1]: I believe two or so tests do not follow the testing naming convention set by the ADR, which should follow either `Test_TheMethodBeingTested_TheCoveredScenario_TheExpectedResult` or `Test_TheMethodBeingTested_TheExpectedResult`

File: src/utils/config_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+	test := Config{}
+
+	err := test.populatePartnerSettings(jsonInput)
+
+	assert.Error(t, err)
+	assert.Contains(t, buffer.String(), "Invalid encoding found")
+}
+
+func Test_validateDefaultEncoding_errors(t *testing.T) {
```
Comment (pluckyswan) [inline src/utils/config_test.go:57]: Is validateDefaultEncoding missing a happy path or is populateConfig covering it enough that we don't need it

File: src/utils/config.go (hunk 1 above)
Comment (halprin) [reply src/utils/config.go:37]: Or rename the original argument a the top to `jsonData`.

File: src/utils/config_test.go (hunk 3 above)
Comment (jcrichlake) [reply src/utils/config_test.go:57]: The latter

**Review by halprin — APPROVED** [review]

File: src/utils/config.go (hunk 2 above)
Comment (jcrichlake) [reply src/utils/config.go:28]: I think for now these settings are okay

**Pull request merged**